File: 2025/JuyoungMoon/week9/2025-08-03-assign-cookies.cpp
// Author: Juyoung Moon

// KDD LeetCode Study Week 9: Greedy
// https://github.com/juyomo/leetcode-study

// LeetCode #455.
// https://leetcode.com/problems/assign-cookies/

class Solution {
public:
    int findContentChildren(vector<int>& kids, vector<int>& cookies) {
        sort(kids.begin(), kids.end());
        sort(cookies.begin(), cookies.end());

        int k = 0;
        int c = 0;
        int satisfied = 0;

        while (k < kids.size() && c < cookies.size()) {
            if (kids[k] <= cookies[c]) {
                satisfied++;
                k++;
            } 
            
            c++;
        }

        return satisfied;
    }
};

File: Chohyeon Kim/2025-08-03-Assign Cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()

        count = 0
        i, j = 0, 0

        while i < len(g) and j < len(s):
            if g[i] <= s[j]:
                count += 1
                i += 1
            j += 1

        return count
